File: frontend/src/api/apiClient.js
import axios from 'axios';

// ============================================================
// DATA ACCESS LAYER - Axios Base Client kết nối .NET Web API
// ============================================================

const API_BASE_URL = import.meta.env.VITE_API_URL || 'https://localhost:7001/api';

export const apiClient = axios.create({
  baseURL: API_BASE_URL,
  headers: {
    'Content-Type': 'application/json'
  }
});

// Tự động gắn Token xác thực JWT (nếu có) vào Header mỗi Request
apiClient.interceptors.request.use((config) => {
  const token = localStorage.getItem('blush_jwt_token');
  if (token) {
    config.headers.Authorization = `Bearer ${token}`;
  }
  return config;
}, (error) => Promise.reject(error));

File: frontend/lib/api/api_client.dart
import 'dart:convert';
import 'package:http/http.dart' as http;

class ApiClient {
  // Base URL kết nối tới Backend ASP.NET Core Web API (.NET 8)
  // Khi chạy trên Android Emulator: dùng http://10.0.2.2:5000/api
  // Khi chạy trên Máy thật: dùng IP Wi-Fi máy chạy .NET (VD: http://192.168.1.5:5000/api)
  static const String baseUrl = 'http://10.0.2.2:5000/api';

  static Map<String, String> get headers => {
        'Content-Type': 'application/json',
        'Accept': 'application/json',
      };

  static Future<http.Response> get(String endpoint) async {
    final url = Uri.parse('$baseUrl/$endpoint');
    return await http.get(url, headers: headers);
  }

  static Future<http.Response> post(String endpoint, Map<String, dynamic> data) async {
    final url = Uri.parse('$baseUrl/$endpoint');
    return await http.post(
      url,
      headers: headers,
      body: jsonEncode(data),
    );
  }
}
